src/d3a/models/area/__init__.py

```diff
@@ -143,6 +145,7 @@ def __init__(self, name: str = None, children: List["Area"] = None,
             if external_connection_available and self.strategy is None else None
         self.should_update_child_strategies = False
         self.external_connection_available = external_connection_available
+        self.total_energy_deviance_kWh: Dict[DateTime, float] = {}
```

Comment:
I think that a much much better place for all these changes (total_energy_deviance member + the method `_calculate_energy_deviances` ) is the AreaStats class. That way the Area class will not bloat more than it already has. You can also handle the updates in the energy deviance without changing the Area class, by adding it in the `AreaStats.update_area_market_stats` method (it is already called every market cycle in order to update other statistics).

Reply:
> I think that a much much better place for all these changes (total_energy_deviance member + the method _calculate_energy_deviances ) is the AreaStats class.

Done, https://github.com/gridsingularity/d3a/pull/1230/commits/9a79f2871e5898b88bfb21096e0e1f7695b8d11a

> You can also handle the updates in the energy deviance without changing the Area class, by adding it in the AreaStats.update_area_market_stats method (it is already called every market cycle in order to update other statistics).

Unfortunately, it need to be called for both cases when area has children or itself is a device. For later case due to [early return](https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/area/__init__.py#L332) for cases when area is a device, and for other non-device area where market creation is done from top-to-bottom where concatinating into `update_area_market_stats` left me child with no deviance calculation due to market creation from top to bottom, so I also have to `calculate_energy_deviances ` before the end of `cycle_market`